### ISSUE TITLE
azurerm_storage_share_file: content_md5 is always detected as changed

### ISSUE THREAD

**Issue by plule-ansys** [hashicorp/terraform-provider-azurerm#10888]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

```
Terraform v0.14.7
+ provider registry.terraform.io/hashicorp/azurerm v2.50.0
```

### Affected Resource(s)

* `azurerm_storage_share_file`

### Terraform Configuration Files

```hcl
provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_storage_account" "example" {
  name                     = "azureteststorage"
  resource_group_name      = azurerm_resource_group.example.name
  location                 = azurerm_resource_group.example.location
  account_tier             = "Standard"
  account_replication_type = "LRS"
}

resource "azurerm_storage_share" "example" {
  name                 = "sharename"
  storage_account_name = azurerm_storage_account.example.name
  quota                = 1
}

resource "azurerm_storage_share_file" "example" {
  name             = "empty"
  storage_share_id = azurerm_storage_share.example.id
  content_md5 = "d41d8cd98f00b204e9800998ecf8427e" # Empty file md5
  source           = "empty"
}
```

### Debug Output

I had issue to get it, and was worried about possible internal things leaking here. Let me know if it can help for this issue, I will take the time to provide it.

### Expected Behaviour

A first call to terraform apply uploads the file, further calls don't.

### Actual Behaviour

The first call correctly uploads the file, but further calls present the `content_md5` as new:
```
  ~ resource "azurerm_storage_share_file" "example" {
      + content_md5      = "d41d8cd98f00b204e9800998ecf8427e"
        id               = "<snip>"
        name             = "empty"
        # (4 unchanged attributes hidden)
    }
```

### Steps to Reproduce

1. `touch empty`
2. `terraform apply`
3. `terraform apply`

### References

No related issue or PR were found.

**Comment by alvintang** [comment]
Hi. I looked a little into this and it seems that the return of GetProperties call in [storage_share_file_resource.go](https://github.com/terraform-providers/terraform-provider-azurerm/blob/2b30abb96725abae3c38e846bf1d8a1e2b0fcf2d/azurerm/internal/services/storage/storage_share_file_resource.go#L289) has an empty `ContentMD5`. I looked further in the Microsoft doc[1] and the function call in Giovanni storage SDK [2]. Giovanni uses response header `x-ms-content-md5` and according to the Microsoft doc, the `Range` or `x-ms-range` header must be set for this response header to be returned:
```
Starting from version 2016-05-31, if the file has a MD5 hash, and if  request contains range header (Range or x-ms-range), this response  header is returned with the value of the whole file’s MD5 value. This  value may or may not be equal to the value returned in Content-MD5  header, with the latter calculated from the requested range.
```
I tested this by replacing `x-ms-content-md5` with `Content-MD5` in this [file](https://github.com/terraform-providers/terraform-provider-azurerm/blob/2b30abb96725abae3c38e846bf1d8a1e2b0fcf2d/vendor/github.com/tombuildsstuff/giovanni/storage/2019-12-12/file/files/properties_get.go#L114) in the giovanni package under the vendor folder. I was able to retrieve the value. Since giovanni is not passing any range header anyway, is this a valid fix? Should this be raised as PR in this repo or in giovanni repo?

References: 
[1] https://docs.microsoft.com/en-us/rest/api/storageservices/get-file
[2] https://github.com/tombuildsstuff/giovanni/blob/9de4e973cdac0d3f7b512b4afaed66393d3c608c/storage/2019-12-12/file/files/properties_get.go#L114

**Comment by daveneeley** [comment]
Now that a giovanni update has been released, what's next to get this in an azurerm release?

**Comment by daveneeley** [comment]
@alvintang @tombuildsstuff I've updated the version of giovanni that includes the change for the bug in the Pull Request. It compiles locally, but I'm not sure how to get it tested. Would you please advise?

**Comment by daveneeley** [comment]
I had to submit a new PR since I did the first one from the wrong branch. It needs a maintainer to allow checks to run. It has a test now thanks to @alvintang .

**Comment by tombuildsstuff** [comment]
I believe this has been fixed by #13020, closing - but please let us know if this is still an issue and we'll take another look.
